tests/boot/with_mcumgr/pytest/test_security.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import pytest
+import logging
+
+from pathlib import Path
+from twister_harness import Device, MCUmgr
+from conftest import (
+    wait_for_message,
+    west_sign_with_imgtool,
+    find_in_config,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    'key_file',
+    [
+        '',
+        'root-ec-p256.pem',
+        'root-rsa-2048.pem'
+    ],
+    ids=[
+        'no_key',
+        'invalid',
+        'valid'
+    ])
+def test_upgrade_signature(dut: Device, mcumgr: MCUmgr, key_file):
+    """
+    Verify that the application can be updated only when app is signed with valid key
+    1) Device flashed with MCUboot and an application that contains SMP server
+    2) Prepare an update of an application containing the SMP server that has
+       been signed:
+       a) without any key
+       b) with a different key than MCUboot was compiled with
+       c) with the same key as MCUboot was compiled with
+    3) Upload the application update to slot 1 using mcumgr
+    4) Flag the application update in slot 1 as 'pending' by using mcumgr 'test'
+    5) Restart the device, verify that:
+      - swap is not started with reason invalid signature and the original
+      application is booted when 2a (no key) and 2b (invalid key)
+      - the updated application is booted when 2c (valid key)
+    """
+    assert wait_for_message(dut, "Launching primary slot application")
+
+    smp_svr_dir = Path(dut.device_config.build_dir) / 'smp_svr'
+    image_to_test = Path(dut.device_config.build_dir) / 'smp_svr.tested.bin'
+    new_key_file = ''
+    valid_key = False
+    if key_file:
+        origin_key_file = find_in_config(
+            Path(dut.device_config.build_dir) / 'mcuboot' / 'zephyr' / '.config',
+            'CONFIG_BOOT_SIGNATURE_KEY_FILE'
```

Comment:
this will not work, sysbuild will forcefully fix the wrong configuration with that from the sysbuild configuration

Reply:
It works. Here I'm looking for the signature key file path in existing build output (not using any wrong `key_file` from pytest parameters).